packages/node/src/evm/fulfillments/index.ts

```diff
@@ -65,7 +65,17 @@ export async function submit(state: ProviderState<EVMProviderState>) {
     return [...submittedApiCalls, ...submittedWithdrawals];
   });
 
-  const responses = await Promise.all(promises);
+  const receipts = (await Promise.all(promises)) as TransactionReceipt[];
+  return receipts;
+}
 
-  return responses;
+export function applyFulfillments<T>(requests: ClientRequest<T>[], receipts: TransactionReceipt[]) {
+  return requests.reduce((acc, request) => {
+    const receipt = receipts.find((r) => r.id === request.id);
```

Comment:
Good point. I was kinda on the fence about this one since it requires a bit more cognitive load from the developer to figure out what the lodash `find()` is doing vs just using the regular JS `.find()` that everyone already knows. I'm going to go back to using just the regular JS `.find()`